bundles/org.openhab.binding.modbus.helioseasycontrols/src/main/java/org/openhab/binding/modbus/helioseasycontrols/internal/HeliosEasyControlsHandler.java

```diff
@@ -0,0 +1,684 @@
+                String payload = v.getVariableString();
+                this.comms.submitOneTimeWrite(new ModbusWriteRegisterRequestBlueprint(
+                        HeliosEasyControlsBindingConstants.UNIT_ID, HeliosEasyControlsBindingConstants.START_ADDRESS,
+                        new ModbusRegisterArray(preparePayload(payload)), true,
+                        HeliosEasyControlsBindingConstants.MAX_TRIES), result -> {
+                            this.comms.submitOneTimePoll(
+                                    new ModbusReadRequestBlueprint(HeliosEasyControlsBindingConstants.UNIT_ID,
+                                            ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS,
+                                            HeliosEasyControlsBindingConstants.START_ADDRESS, v.getCount(),
+                                            HeliosEasyControlsBindingConstants.MAX_TRIES),
+                                    pollResult -> {
+                                        lock.release();
+                                        Optional<ModbusRegisterArray> registers = pollResult.getRegisters();
+                                        if (registers.isPresent()) {
+                                            processResponse(v, registers.get());
+                                        }
+                                    }, failureInfo -> {
+                                        String errorMsg = failureInfo.getCause().getMessage();
+                                        logger.warn("{} encountered error reading from device: {}",
+                                                HeliosEasyControlsHandler.class.getSimpleName(), errorMsg);
+                                        lock.release();
+                                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                                errorMsg);
+                                    });
+                        }, failureInfo -> {
+                            String errorMsg = failureInfo.getCause().getMessage();
+                            logger.warn("{} encountered error writing to device: {}",
+                                    HeliosEasyControlsHandler.class.getSimpleName(), errorMsg);
+                            lock.release();
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+
+                        });
+            }
+        } else {
+            if (this.comms == null) {
+                this.handleError("Modbus communication interface is unavailable.",
+                        ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            if (this.vMap == null) {
+                this.handleError("Variable definition is unavailable.", ThingStatusDetail.CONFIGURATION_ERROR);
+            }
+        }
+    }
+
+    // TODO: set time incl. timezone, update time channels based on timezone read from device

```

Comment:
Can this be removed?

Reply:
Still something I intend to implement...

Reply:
In general code to be commited to the official repo shouldn't contain any TODOs. You could open an issue, if you don't want to implement it in this round (which would be best according the 2.5.9 merge window I think).